app/controllers/application_controller.rb

```diff
@@ -1,10 +1,39 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
+    enable :sessions
+    set :session_secret, 'fwitter'
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  helpers do
+
+    def logged_in?
+      !!session[:user]
+    end
+
+    def current_user
```

Comment:
That makes sense!
Was a bitch to refactor, had to hunt down every time current_user was called plus a whole bunch of other bugs that came up, but yeah, i got it in the end :)